pkgs/development/python-modules/awkward/default.nix

```diff
@@ -1,40 +1,35 @@
 { lib
 , buildPythonPackage
 , fetchPypi
-, cmake
+, awkward-cpp
+, hatch-fancy-pypi-readme
+, hatchling
 , numba
 , numpy
 , pytestCheckHook
 , pythonOlder
-, pyyaml
-, rapidjson
-, setuptools
 }:
 
 buildPythonPackage rec {
   pname = "awkward";
-  version = "1.10.2";
-  format = "setuptools";
+  version = "2.0.0";
+  format = "pyproject";
 
   disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-MDvAkZ8JMts+eKklTBf83rEl5L5lzYlLQN+8O/3fwFQ=";
+    hash = "sha256-N4KzRkMIPW7nZE6f2z2ur8S2AwpmfyGf1hy3sjSXa2g=";
   };
 
   nativeBuildInputs = [
-    cmake
-  ];
-
-  buildInputs = [
-    pyyaml
-    rapidjson
+    hatch-fancy-pypi-readme
+    hatchling
   ];
 
   propagatedBuildInputs = [
+    awkward-cpp
     numpy
-    setuptools
```

Comment:
According to [this file](https://github.com/scikit-hep/awkward/blob/main/pyproject.toml#L42), `packaging` is needed and so is `typing-extensions` on Python < 3.11.